### ISSUE TITLE
[Windows10] Dark age of Camelot directx9 32 bit game

### ISSUE THREAD

**Issue by Filters83** [doitsujin/dxvk#3666]
Hello i tested latest versione and 1.10 version, the game run super bad whit both version. Heavy swapping and the game is unplayable. The game is using Directx9 32bit. I usually play on this server since is free ( https://www2.uthgard.net ) that use the base game patched from them. 
The reason i tested this Vulkan api system is casue even on normal the game run super bad whit this graphic card even disabling ULPS ( ultra low power settings ), basicly it will not use the gpu at all and constantly downclocking. Was hoping Vulkan can solve this problem

### Software information
Dark Age of camelot base settings bordeless windows at monitor resolution 1440p

### System information
- GPU: Amd Rx6700XT
- Driver: 23.8.2
- Wine version: 
- DXVK version: 2.3 and 1.10 tested

**Comment by Blisto91** [comment]
Do you mean the game have low fps in general with dxvk or that there is alot of stuttering?
I tried the game on the main servers quickly and both with my 7900xtx dGPU and weak zen4 iGPU my fps seems to always be at 125 fps in the starting area i choose.

The game does have quite alot of shaders and they will get compiled when you move and look around in the game which will introduce alot of stutter (should get better over time),b ut dxvk can't do much about that. That part should improve once AMD's driver supports the `VK_EXT_graphics_pipeline_library` Vulkan extension which is utilized to help such issues. The radv driver on Linux (not AMD's own) already support this, though even without it fares better in regards to shader compilation.
Though i also see game always loads them in at draw time and not up front.

**Comment by Filters83** [comment]
Yes stuttering i also get 120 fps but is the cap, the problem lies in the shader. I found a discussione on reddit some minute ago ( https://www.reddit.com/r/Amd/comments/9mpzvj/is_there_anyway_to_switch_global_shader_cache_to/ ) enabling shader cache to on greatly improved in dx9. Ill try use vulkan again to see if it goes even better. Btw im having a Ryzen 5700G Cpu and whit that integrated gpu the game was working no problem even on dx9 so i guess its something wrong whit just the dedicated graphic card even if they use the same driver ...

**Comment by K0bin** [comment]
Please make an apitrace so I can look into this.

**Comment by Blisto91** [comment]
Native Windows trace https://drive.proton.me/urls/QR9QH5383G#SwRs-2020qF8

**Comment by tdm4** [comment]
@Filters83 I use dxvk to play on the Dark Age of Camelot freeshard because using the normal 'gl' renderer causes a massive memory leak (in libMesa somewhere). I switched renderer=vulkan and the game plays fine. Once in a while I do get a crash to desktop but it seems stable to me.

How did you manage to get Uthgard freeshard to run? When I try, I see game.dll running and connecting to the server, but there is no graphics.

**Comment by Blisto91** [comment]
This has now been fixed by the new ubershader feature in the 3.0 release. Seems very smooth after the initial compile from some short testing.
